pkg/proxy/proxier.go

```diff
@@ -319,38 +319,43 @@ type Proxier struct {
 	hostIP        net.IP
 }
 
+var (
+	ErrProxyOnLocalhost = fmt.Errorf("cannot proxy on localhost")
```

Comment:
So then we have to expose a method to check it.  Any everyone has to check it the same way.  And everyone needs a consistent message.  Seems like you're arguing against people returning known error types from functions?  If I set up a TLS config and get back a TLS typed error (this is invalid) isn't that reasonable?

----- Original Message -----
> > @@ -319,38 +319,43 @@ type Proxier struct {
> >  	hostIP        net.IP
> >  }
> >  
> > +var (
> > +	ErrProxyOnLocalhost = fmt.Errorf("cannot proxy on localhost")
> 
> Shouldn't you pre-validate that, then, rather than surfacing it from the
> depths of the app?  It's a pretty specific error to catch, you might as well
> check it yourself.
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/GoogleCloudPlatform/kubernetes/pull/8269/files#r30387927